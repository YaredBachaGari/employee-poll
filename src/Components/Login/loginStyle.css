.login-container {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  text-align: center;
}
.input-container {
  width: 100%;
  margin: 0.25rem auto;
}
.input-container input {
  display: block;
  width: 90%;
  height: 2rem;
  margin: 0rem auto;
  border-radius: 0.5rem;
  border: 0.5px solid #737272;
  font-size: 1rem;
}
.login-btn button{
    font-size: 1rem;
    font-weight: bold;
    padding:0.5rem 2rem;
    margin-top: 1.5rem;
    background-color:#eeeeee;
    border: 1px solid #737272;
    border-radius: 0.5rem;
}

.login-btn button:hover{
    background-color:brown; 
    color: #eeeeee;
}