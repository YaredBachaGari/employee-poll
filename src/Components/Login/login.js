import React, { useState } from "react";
import "./loginStyle.css";
import { useNavigate, Navigate } from "react-router-dom";
import { AuthenticateUserSuccess } from "../../Redux-handler/Actions/AuthUser";
import { useDispatch } from "react-redux";

const Login = ({ UserData, Auth }) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const redirectPath = localStorage.getItem("redirectPath");
  const [userInfo, setUserInfo] = useState({ username: "", password: "" });
  const [error, setError] = useState("");
  const data = UserData?.data;
  const onchangeHandler = (e) => {
    e.preventDefault();
    const name = e.target.name;
    const value = e.target.value;
    setUserInfo((prev) => {
      return {
        ...prev,
        [name]: value,
      };
    });
  };

  const submitHandler = (e) => {
    e.preventDefault();
    let foundUser = false;
    let user;
    if (userInfo?.username && userInfo?.password) {
      for (let key in data) {
        if (
          data[key]?.password === userInfo?.password &&
          key === userInfo?.username
        ) {
          const { id, name, avatarURL } = data[key];
          user = { id, username: key, name, avatarURL };
          ///navigate("/home");
          if (redirectPath) {
            //localStorage.removeItem("redirectPath");
             //<Navigate to={redirectPath} replace />;
             navigate(redirectPath)
          } else {
            navigate('/home')
             //<Navigate to="/home" replace />;
          }
          setUserInfo({ username: "", password: "" });
          foundUser = true;
          break;
        }
      }
      if (!foundUser) {
        setError("Incorrect Password or userName");
      }
    } else {
      setError("No Password or userName entered");
    }
    if (foundUser && user) {
      dispatch(AuthenticateUserSuccess(user));
    }
  };

  return (
    <div className="login-container">
      {error && <p style={{ color: "red" }}>{error}!!!</p>}
      <div className="input-container">
        <label htmlFor="username">User Name</label>
        <input
          id="username"
          value={userInfo.username}
          name="username"
          placeholder="username"
          onChange={onchangeHandler}
          data-testid="username"
        />
      </div>
      <div className="input-container">
        <label htmlFor="password">Password</label>
        <input
          id="password"
          type="password"
          value={userInfo.password}
          name="password"
          placeholder="Password"
          onChange={onchangeHandler}
          data-testid="password"
        />
      </div>
      <div className="login-btn">
        <button data-testid="loginbtn" className="btn" onClick={submitHandler}>
          Submit
        </button>
      </div>
    </div>
  );
};

export default Login;
