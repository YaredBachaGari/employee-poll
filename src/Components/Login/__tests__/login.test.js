import React from "react";
import { render, screen } from "@testing-library/react";
import Login from "../login.js";
import { BrowserRouter } from "react-router-dom";
import store from "../../../Redux-handler/Store/Store";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";

describe("should render avatar", () => {
  it("the snapshoots must match the login form and all fields should exist", () => {
    const { login } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    const userName = screen.getByTestId("username");
    const passcode = screen.getByTestId("password");
    expect(login).toMatchSnapshot();
    expect(userName).toBeInTheDocument();
    expect(passcode).toBeInTheDocument();
  });
});
