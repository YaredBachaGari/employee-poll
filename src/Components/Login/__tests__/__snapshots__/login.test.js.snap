// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render avatar the snapshoots must match the login form and all fields should exist 1`] = `undefined`;
